### ISSUE TITLE
Animate time-dynamic layer with user-defined start time, stop time, and time step

### ISSUE THREAD

**Issue by rsignell-usgs** [TerriaJS/terriajs#984]
There are several issues surrounding the handling maps with a time dimension.   In https://github.com/TerriaJS/terriajs/issues/889 the default time value was modified, in  https://github.com/TerriaJS/terriajs/issues/942#issuecomment-149234052, @metanomy asks for more interaction with the time slider, and in https://github.com/TerriaJS/terriajs/issues/927, there is a enhancement request for a godiva2 like client.

A couple of problems solved by the godiva2 client are:
1) sometimes animating the entire dataset over time takes a very long time (e.g we have a 10TB dataset served via WMS), and is not what the user wants
2) sometimes the time resolution is more refined than the user wants (e.g. the user might want to animate sea surface temperature once-per-day instead of every hourly time step)

When godiva2 loads a time dependent dataset, the default is to display the time step that is closest to the current time.   The user can then select a different timestep by picking a date off the calendar and picking a time-of-day from the drop-down menu:
![2015-10-19_10-49-53](https://cloud.githubusercontent.com/assets/1872600/10581599/8ef90654-7651-11e5-89b5-f983c0e163ed.png)

If the user than wants to select an animation, they click on "first frame", select a time (using the same calendar-and-drop-down-menu method) and then click on "last frame" (again, using the same method).   They then get a menu that offers different choices of time resolution for the animation, along with a suggestion that they use the smallest number they need (for speed):
![2015-10-19_10-55-42](https://cloud.githubusercontent.com/assets/1872600/10581681/f965bc94-7651-11e5-86b9-f284b8afd32f.png)

TerriaJS might thing about adding similar functionality, as it is dramatically improves the experience for time animation of large 3D and 4D datasets.

**Comment by metanomy** [comment]
+1

**Comment by stevage** [comment]
@AlexGilleran is working on this atm. Alex, have a look at this and feel free to close if you've either supported this or decided not to?

**Comment by AlexGilleran** [comment]
Nah, what I've done supports gaps in data, there's no option for the user to limit the steps in between data.
